src/EA.Weee.RequestHandlers/Admin/AatfReports/GetAllAatfSentOnDataCsvHandler.cs

```diff
@@ -0,0 +1,109 @@
+            this.csvWriterFactory = csvWriterFactory;
+        }
+        public async Task<CSVFileData> HandleAsync(GetAllAatfSentOnDataCsv request)
+        {
+            authorization.EnsureCanAccessInternalArea();
+            if (request.ComplianceYear == 0)
+            {
+                var message = $"Compliance year cannot be \"{request.ComplianceYear}\".";
+                throw new ArgumentException(message);
+            }
+            PanArea panArea = null;
+            UKCompetentAuthority authority = null;
+            if (request.AuthorityId != null)
+            {
+                authority = await commonDataAccess.FetchCompetentAuthorityById(request.AuthorityId.Value);
+            }
+            if (request.PanArea != null)
+            {
+                panArea = await commonDataAccess.FetchLookup<PanArea>(request.PanArea.Value);
+            }
+
+            var obligatedData = await weeContext.StoredProcedures.GetAllAatfSentOnDataCsv(request.ComplianceYear, request.AATFName, request.ObligationType, request.AuthorityId, request.PanArea);
+
+            //Remove the Id columns
+            if (obligatedData.Tables.Count > 0 && obligatedData.Tables[0] != null)
+            {
+                if (obligatedData.Tables[0].Columns.Contains("AatfId"))
+                {
+                    obligatedData.Tables[0].Columns.Remove("AatfId");
+                }
+                if (obligatedData.Tables[0].Columns.Contains("ReturnId"))
+                {
+                    obligatedData.Tables[0].Columns.Remove("ReturnId");
+                }
+                if (obligatedData.Tables[0].Columns.Contains("Q"))
+                {
+                    obligatedData.Tables[0].Columns.Remove("Q");
+                }
+                if (obligatedData.Tables[0].Columns.Contains("CategoryId"))
+                {
+                    obligatedData.Tables[0].Columns.Remove("CategoryId");
+                }
+                if (obligatedData.Tables[0].Columns.Contains("TonnageType"))
+                {
+                    obligatedData.Tables[0].Columns.Remove("TonnageType");
+                }
+            }
+
+            var fileName = string.Format("{0}", request.ComplianceYear);
+            if (request.AuthorityId != null)
+            {
+                fileName += "_" + authority.Abbreviation;
+            }
+            if (request.PanArea != null)
+            {
+                fileName += "_" + panArea.Name;
+            }
+            if (request.AATFName != null)
+            {
+                fileName += "_" + request.AATFName;
+            }
+            if (request.ObligationType != null)
+            {
+                fileName += "_" + request.ObligationType;
+            }
+
+            fileName += string.Format("_AATF WEEE sent on for treatment_{0:ddMMyyyy_HHmm}.csv",
+                                DateTime.UtcNow);
+
+            string fileContent = string.Empty;
+            if (obligatedData.Tables.Count > 0)
```

Comment:
Add test that if no are no tables that the file content should be string.empty

Reply:
Added test